File: Python/7py 2/solution_cats.py
"""
|--------------------------------------------------------------------------
| Cats and shelves
|--------------------------------------------------------------------------
|
"""

def solution(start, finish):  
   diff = finish-start
   return ((diff / 3) + (diff % 3))//1


def solution(start, finish):
   count = 0
   while start != finish:
      if start + 3 <= finish: start +=3
      else: start +=1
      count += 1
   return count


def solution(start, finish):
   return sum(divmod(finish-start, 3))


print(solution(1,5), 2)
print(solution(2,4), 2)

File: PHP/7php 3/solution_cats.php
<?php
/*
|--------------------------------------------------------------------------
| Cats and shelves
|--------------------------------------------------------------------------
|
*/

function cats(int $start, int $finish): int {
   $diff = $finish - $start;
   return floor($diff / 3) + ($diff % 3);
}


function cats($s,$f) {
   return intdiv($f-$s,3)+($f-$s)%3;
}


function cats(int $start, int $finish): int {
   $count =0 ;
   $diff = $finish - $start;
   while($diff>0) {
      if($diff == 1 || $diff == 2) {
         $diff -= 1;
      } else if ($diff >= 3) {
         $diff -= 3;
      }
      $count++;
   }
   return $count;
}


print_r(cats(1,5) . ' ' . 2 . '
');
print_r(cats(2,4) . ' ' . 2 . '
');